go-controller/pkg/config/config.go

```diff
@@ -406,32 +424,22 @@ func InitConfig(ctx *cli.Context, defaults *Defaults) error {
 		}
 		logrus.Infof("Parsed config file %s", f.Name())
 		logrus.Infof("Parsed config: %+v", cfg)
-
-		if defaults != nil {
-			fillDefaults(&cfg, defaults)
-		}
 	}
 
-	// command-line > config file > default values
-	Default.MTU = getConfigInt(cliConfig.Default.MTU, cfg.Default.MTU, 1400)
-	Default.ConntrackZone = getConfigInt(cliConfig.Default.ConntrackZone, cfg.Default.ConntrackZone, 64000)
-
-	Logging.File = getConfigStr(cliConfig.Logging.File, cfg.Logging.File, "/var/log/ovn-kubernetes.log")
-	Logging.Level = getConfigInt(cliConfig.Logging.Level, cfg.Logging.Level, 4)
-
-	CNI.ConfDir = getConfigStr(cliConfig.CNI.ConfDir, cfg.CNI.ConfDir, "/etc/cni/net.d")
-	CNI.Plugin = getConfigStr(cliConfig.CNI.Plugin, cfg.CNI.Plugin, "ovn-k8s-cni-overlay")
+	if defaults == nil {
+		defaults = &Defaults{}
+	}
 
-	if err = validateKubernetesConfig(&cliConfig, &cfg); err != nil {
+	if err = buildKubernetesConfig(&cliConfig, &cfg, defaults); err != nil {
 		return err
 	}
 
-	OvnNorth.ClientAuth, OvnNorth.ServerAuth, err = getOvnAuth(true, &cliConfig.OvnNorth, &cfg.OvnNorth)
+	OvnNorth, err = buildOvnAuth("nb", &cliConfig.OvnNorth, &cfg.OvnNorth, defaults.OvnNorthAddress)
```

Comment:
The returned value is not being used anywhere. I am missing something here.

Reply:
I also see that the logging from cliconfig is not being copied into config. So any setting to CLI won't work. For e.g., -logfile is not being honored right now.

Reply:
> The returned value is not being used anywhere. I am missing something here.

@shettyg OvnNorth is a global defined near the top.

Reply:
> I also see that the logging from cliconfig is not being copied into config

@shettyg will fix